index.js

```diff
@@ -40,6 +34,35 @@ module.exports = function (src, dest, opts) {
 		read.on('end', function () {
 			resolve(read);
 		});
+	});
+};
+
+function statSource(src, progress) {
+	if (progress.enabled) {
+		return fsP.stat(src).then(function(stat) {
+			progress.stat = stat;
+		}).catch(function(err) {
+			Promise.reject(new CpFileError('NO_ENTRY', err));
+		});
+	}
+	return Promise.resolve();
+};
+
+module.exports = function (src, dest, opts) {
+	if (!src || !dest) {
+		return Promise.reject(new CpFileError('`src` and `dest` required'));
+	}
+
+	opts = objectAssign({overwrite: true, onProgress: null}, opts);
```

Comment:
I would prefer the [null pattern](https://en.wikipedia.org/wiki/Null_Object_pattern), in this case:
```js
opts = objectAssign({overwrite: true, onProgress: function() {}}, opts);
```
Thus you can drop all `if (progress.enabled)`, which improves the readability.